src/gennylib/src/PoolManager.cpp

```diff
@@ -62,9 +74,13 @@ mongocxx::pool::entry genny::v1::PoolManager::client(const std::string& name,
 
     Pools& pools = lap.second;
 
-    auto& pool = pools[instance];
+    if (!pools.encryption) {
+        pools.encryption = setupEncryption(name, context, _dryRun);
+    }
+
+    auto& pool = pools.instances[instance];
     if (pool == nullptr) {
-        pool = createPool(name, this->_apmCallback, context);
+        pool = createPool(name, this->_apmCallback, pools.encryption, context);
```

Comment:
the context is being passed to createPool, would it not make more sense to let it create and configure encryption?

Reply:
The reason that `setupEncryption()` is being done here, instead of within `createPool()` is that the initialization that's being done in `setupEncryption()` (e.g. creating the key vault & data keys) must only happen once, when the very first "instance" of this Pool is created.

When subsequent instances of this pool name are initialized, they would need to reuse the existing EncryptionContext to set up their auto-encryption options.  These subsequent instances cannot just re-create the EncryptionContext from the YAML node, because if they did, they will lose the data key IDs that were created during `setupKeyVault()`.

Reply:
So you're saying encryption settings are global? I think we could still create this inside createPool. We have a few instances in Genny of creating static vars that we can expect to be reused across different calls, e.g. [here](https://github.com/mongodb/genny/blob/d5e219bbbf7d182e8872abe7df40d082e7cb0ca8/src/metrics/include/metrics/v2/event.hpp#L117).

I'm imagining something like the following inside `createPool`:

    static auto encryption = setupEncryption(name, context, _dryRun);

I think it would be worth doing that to keep this construction internal to the pool construction, and avoid some of the pointer ownership issues / var piping mentioned further up.

Reply:
Also, if it's only the results of `setupKeyVault()` specifically that need to be shared, maybe that could be constructed statically and everything else still treated as pool-specific?